app/src/main/java/org/phenoapps/verify/LoaderDBActivity.java

```diff
@@ -135,16 +126,18 @@ private void parseHeaders(Uri data) {
 
         try {
             //query file path type
-            mFilePath = getPath(mFileUri);
-            int lastDot = mFileUri.toString().lastIndexOf(".");
+//            mFilePath = getPath(LoaderDBActivity.this ,mFileUri);
+            UriHandler uriWorker = new UriHandler();
+            String mFilePath = uriWorker.getPath(LoaderDBActivity.this, mFileUri);
```

Comment:
mFilePath eventually makes it's way to MainActivity, which is then used to populate a default file name for the export file. This creates really weird names when importing from the cloud. Instead just make the default file name something like Verify_timestamp.csv